src/android/DialOnceBridge.java

```diff
@@ -1,68 +1,94 @@
 package com.dialonce.sdk;
 
+import org.apache.cordova.CallbackContext;
 import org.apache.cordova.CordovaPlugin;
+import org.apache.cordova.LOG;
+import org.json.JSONArray;
 import org.json.JSONException;
 
 import android.Manifest;
 import android.content.Context;
 import android.content.Intent;
 import android.content.pm.ApplicationInfo;
 import android.content.pm.PackageManager;
+import android.os.Build;
+import android.text.TextUtils;
 
 import com.dialonce.sdk.DialOnce;
 
 public class DialOnceBridge extends CordovaPlugin {
 
-    String [] permissions = { Manifest.permission.INTERNET,
+    private static final String TAG = "dialonce-cordova";
+    private static final String[] permissions = { Manifest.permission.INTERNET,
             Manifest.permission.ACCESS_NETWORK_STATE,
             Manifest.permission.CALL_PHONE,
             Manifest.permission.PROCESS_OUTGOING_CALLS,
             Manifest.permission.READ_PHONE_STATE,
             Manifest.permission.VIBRATE,
             Manifest.permission.GET_TASKS,
             Manifest.permission.INSTALL_SHORTCUT };
-    private static int REQUEST_CODE_PERMISSION = 0;
+    private static final int REQUEST_CODE_PERMISSION = 0;
 
-    @Override protected void pluginInitialize() {
-        this.initDialOnce();
-    }
+    @Override 
+    protected void pluginInitialize() {
+        try {
+            Context context = cordova.getActivity().getApplicationContext();
+            ApplicationInfo applicationInfo = context.getPackageManager().getApplicationInfo(
+                    context.getPackageName(), PackageManager.GET_META_DATA);
 
-    @Override public void onStart() {
-        //We also initialize Dial Once here just in case it has died. If Dial Once is already set up, this won't do anything.
-        this.initDialOnce();
-    }
+            String apiKey = applicationInfo.metaData.getString("DIALONCE_API_KEY");
```

Comment:
minor: any chance we can make DIALONCE_API_KEY into DIAL_ONCE_API_KEY ? just to separate words in company name